### PR TITLE
[Storage] [WebJobs] Added support for `BlobsOptions.MaxDequeueCount`

### LINKED ISSUE #34269: [Bug] [FEATURE REQUEST] Allowing option to specify the MaxDequeueCount to Queues underlying in Blobs
### Library name and version

Microsoft.Azure.WebJobs.Extensions.Storage.Blobs

### Query/Question

Related to https://github.com/Azure/azure-sdk-for-net/issues/33672

For poison blobs when blob trigger fails it retries 5 times by default.
https://learn.microsoft.com/en-us/azure/azure-functions/functions-bindings-storage-blob-trigger?tabs=in-process&pivots=programming-language-csharp#poison-blobs

However in v4 when the packages were shipped together, customers were able to specify the `MaxDequeueCount` in order to specify the retry amount before the message is added to the blob trigger poison.

Since starting v5, the host.json only applies to queue triggers, and not blob triggers the following properties (including MaxDequeueCount have not been carried forward.

https://learn.microsoft.com/en-us/azure/azure-functions/functions-bindings-storage-queue?tabs=in-process%2Cextensionv5%2Cextensionv3&pivots=programming-language-csharp#host-json

We need to figure out a way to expose this value without using the queues trigger settings.

### Environment

_No response_

### PR DESCRIPTION
**Pull request by amnguye**
Resolves #34269 for the  Microsoft.Azure.WebJobs.Extensions.Storage.Blobs side.

Adds `BlobOptions.MaxDequeueCount` and then transfers the property over when it converts it to the underlying `QueuesOptions`.

Also fixed the samples as well.

### CODE REVIEW

File: sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Config/BlobsOptions.cs [hunk 1]
```diff
@@ -43,6 +46,30 @@ public int MaxDegreeOfParallelism
...
+        /// there are no corresponding poison queues for any queue whose name already ends in "-poison" or any queue
+        /// whose name is already too long to add a "-poison" suffix.
+        /// </remarks>
+        public int MaxDequeueCount
+        {
+            get { return _maxDequeueCount; }
+
+            set
+            {
+                if (value < 1)
```
Comment (liliankasem) [inline sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Config/BlobsOptions.cs:64]: Are there any limits on the max value this can be?

**Review by liliankasem — APPROVED** [review]
lgtm! If there's a private build somewhere, I'm happy to help validate this E2E with the host

File: sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Config/BlobsOptions.cs (hunk 1 above)
Comment (amnguye) [reply sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Config/BlobsOptions.cs:64]: Not sure, in the QueuesOptions there wasn't a limit either.

Comment (amnguye) [reply sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Config/BlobsOptions.cs:64]: I don't believe there's a limit here, I guess it's Int64.MaxValue?

**Comment by amnguye** [comment]
Live tests failures are unrelated to this PR.

**Review by jaschrep-msft — APPROVED** [review]

**Pull request merged**